dfastmi/kernel/BedLevelCalculator.py

```diff
@@ -0,0 +1,144 @@
+
+    def get_element_wise_maximum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise maximum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmax : numpy.ndarray
+            Maximum bed level change.
+        """
+        return numpy.maximum.reduce(dzb)
+
+    def get_element_wise_minimum(self, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the element wise minimum from the given bed levels.
+
+        Arguments
+        ---------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzmin : numpy.ndarray
+            minimum bed level change.
+        """
+        return numpy.minimum.reduce(dzb)
+
+    def get_linear_average(self, number_of_days: Vector, dzb: List[numpy.ndarray]) -> numpy.ndarray:
+        """
+        This method gets the linear average from the given bed levels.
+
+        Arguments
+        ---------
+        number_of_days : Vector
+            A vector of periods indicating the number of days during which each discharge applies.
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+
+        Returns
+        -------
+        dzgem : numpy.ndarray
+            Yearly mean bed level change.
+        """
+        dzgem = 0
+        for i in range(self.number_of_periods):
+            dzgem += dzb[i] * (number_of_days[i] + number_of_days[i-1]) / 2
+        return dzgem
+
+    def get_bed_level_changes(self, dzq: List[numpy.ndarray], rsigma: Vector) -> List[numpy.ndarray]:
+        """
+        This routine computes the bed level changes.
+        This routine requires that dzq and rsigma have the same length.
+        A stagnant period can be represented by a period with rsigma = 1.
+
+        Arguments
+        ---------
+        dzq : List[numpy.ndarray]
+            A list of arrays containing the equilibrium bed level change for each respective discharge period.
+        rsigma : Vector
+            A tuple of relaxation factors, one for each period.
+
+        Returns
+        -------
+        dzb   : List[numpy.ndarray]
+            List of arrays containing the bed level change at the beginning of each respective discharge period.
+        """
+        vsigma = self.__compute_vsigma__(rsigma, dzq)
+        den = self.__compute_denominator__(self.number_of_periods, vsigma)
+        dzb = self.__compute_dzb_at_the_beginning_of_each_period__(dzq, vsigma, den)
+        return dzb
+
+    def __compute_vsigma__(self, rsigma : List[float], dzq):
```

Comment:
What is the type of `dzq`?

Reply:
List[numpy.ndarray], but since it is a private method I felt that typehinting was less needed since this should already be passed into the parent method correctly.
The rsigma is typehinted because a ceirtain List functionality is needed (I think because of append)

Reply:
discussed: Will be further discussed in dfast-101, can be update for this now.

Reply:
done